src/Elastic.Apm/Metrics/MetricsProvider/GcMetricsProvider.cs

```diff
@@ -201,32 +187,26 @@ public void Dispose()
 
 		private void ClrOnGCHeapStats(GCHeapStatsTraceData a)
 		{
-			if (a.ProcessID == _currentProcessId)
+			if (!_isMetricAlreadyCaptured)
 			{
-				if (!_isMetricAlreadyCaptured)
-				{
-					lock (_lock)
-						_isMetricAlreadyCaptured = true;
-				}
-				_gen0Size = (ulong)a.GenerationSize0;
-				_gen1Size = (ulong)a.GenerationSize1;
-				_gen2Size = (ulong)a.GenerationSize2;
-				_gen3Size = (ulong)a.GenerationSize3;
+				lock (_lock)
+					_isMetricAlreadyCaptured = true;
 			}
+			_gen0Size = (ulong)a.GenerationSize0;
+			_gen1Size = (ulong)a.GenerationSize1;
+			_gen2Size = (ulong)a.GenerationSize2;
+			_gen3Size = (ulong)a.GenerationSize3;
 		}
 
-		private void RuntimeGCEnd(TraceProcess traceProcess, TraceGC gc)
+		private void RuntimeGCEnd(TraceProcess a, TraceGC gc)
```

Comment:
```suggestion
		private void RuntimeGCEnd(TraceProcess _, TraceGC gc)
```

If we rename, then use `_` to indicate it's unused.